### PR TITLE
WIP: Upgrade pulsar-client-go to use OrderingKey

### PR DESCRIPTION
**Pull request by sunxiaoguang**
Upgrade pulsar-client-go to use OrderingKey to choose partition which is
the recommended way for routing message in pulsar.

Signed-off-by: Xiaoguang Sun <sunxiaoguang@zhihu.com>

### What problem does this PR solve? 
Pulsar recommend using OrderingKey for routing if key shouldn't be used instead.  

### What is changed and how it works?
Upgrade pulsar to 0.4.0 which implemented OrderingKey feature that existed in clients of other languages. By encoding partition into OrderingKey, there is no need to use magic name to store partition in Properties.  

### Check List 

Tests 

 - Unit test

Code changes
NA

Side effects
NA

Related changes

### Release note

- No release note

### CODE REVIEW

File: cdc/sink/producer/pulsar/option.go [hunk 1]
```diff
@@ -30,8 +30,6 @@ type Option struct {
 	producerOptions *pulsar.ProducerOptions
 }
 
-const route = "$route"
```
Comment (sunxiaoguang) [inline cdc/sink/producer/pulsar/option.go:33]: Magic property name is no longer needed.

**Comment by sunxiaoguang** [comment]
Looks like there are runtime conflict introduced by the new version of pulsar-client-go need to be fixed.

**Comment by hi-rustin** [comment]
PR currently has a large number of conflicts and I will close it for now. Please feel free to reopen it if you want to continue working on it subsequently. Thank you for your contribution!

**Pull request closed without merging**